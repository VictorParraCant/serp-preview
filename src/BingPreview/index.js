import React from 'react'
import styled from 'styled-components'
import {default as PT} from 'prop-types'

const Wrapper = styled.div`
  font: 13px/normal system-ui,sans-serif;
  text-align: left;
  color: #666;
  max-width: 600px;
`
const Title = styled.h3`
  font-weight: normal;
  font-size: 20px;
  line-height: 24px;
  margin: 0;
  padding: 0;
`
const Description = styled.p`
  font-size: 13px;
  line-height: 20px;
  color: inherit;
  margin: 0;
  padding: 0;

  & em {
    font-weight: bold;
    font-style: normal;
  }
`
const Url = styled.cite`
  padding-bottom: 1px;
  line-height: 20px;
  font-size: 14px;
  word-wrap: break-word;
  font-style: normal;
  color: #006d21;
  display: block;

  &::after {
    content: "";
    display: inline-block;
    vertical-align: middle;
    width: 0; 
    height: 0; 
    border-left: 4px solid transparent;
    border-right: 4px solid transparent;
    border-top: 4px solid currentColor;
    margin-left: 5px;
  }
`
const Link = styled.a`
  text-decoration: none;
  color: #003dbf;
  outline: 0;

  &:hover {
    text-decoration: underline;
  }
`

const BingPreview = ({ title, description, url, link }) =>
  <Wrapper>
    <Title>
      <Link href={link}>{title}</Link>
    </Title>
    <Url>{url}</Url>
    <Description>{description}</Description>
  </Wrapper>

BingPreview.defaultProps = {
  description: null,
  link: '#'
}

BingPreview.propTypes = {
  /** Title */
  title: PT.string.isRequired,
  /** Meta Description */
  description: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]),
  /** Url */
  url: PT.string.isRequired,
  /** Link for src */
  link: PT.string
}

export default BingPreview
