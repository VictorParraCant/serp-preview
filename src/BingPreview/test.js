import React from 'react'
import BingPreview from '.'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

it('show preview', () => {
  const tree = renderer
    .create(
      <BingPreview
        title="test title"
        metadescription="meta description"
        url="test.com" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
