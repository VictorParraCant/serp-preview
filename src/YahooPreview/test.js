import React from 'react'
import BingPreview from '.'
import renderer from 'react-test-renderer'

it('show preview', () => {
  const tree = renderer
    .create(
      <BingPreview
        title="test title"
        metadescription="meta description"
        url="test.com" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
