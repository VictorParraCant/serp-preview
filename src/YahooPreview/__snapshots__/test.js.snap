// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show preview 1`] = `
<div
  className="sc-AxjAm dRUbXm"
>
  <h3
    className="sc-AxirZ epSLMX"
  >
    <a
      className="sc-AxhUy dlFYfN"
      href="#"
    >
      test title
    </a>
  </h3>
  <cite
    className="sc-AxhCb eMDuwR"
  >
    test.com
  </cite>
  <p
    className="sc-AxiKw cGRTAA"
  />
</div>
`;
